### DISCUSSION TITLE
Help creating MOTIS server

### DISCUSSION THREAD

**Discussion by NishilJ** [motis-project/motis#642] [Q&A] [answered]
I am trying to create a MOTIS server on ubuntu. This is what I entered when I ran it.
./motis --import.paths osm:dallas_texas.osm.pbf schedule-dal:dart.gtfs.zip
This is the result.
![image](https://github.com/user-attachments/assets/f27026e8-d2c5-4cb9-b3a3-06dd82d45af2)

**Comment by felixguendling** [comment]
Could be out of memory. What are the specs of your machine?
Without the links to your datasets, I cannot reproduce you issue.

**Comment by NishilJ** [comment]
Thank you for responding. I am using a very cheap vps, only 1gb ram and 1 core. I am guessing this is not enough for MOTIS. What is the recommended amount of memory. For GTFS I use: https://www.dart.org/about/about-dart/fixed-route-schedule and for OSM data I get from https://www.interline.io/osm/extracts/ for Dallas, TX, USA. 

When I run on my personal PC (much better spec) with WSL I get similar problem.
![image](https://github.com/user-attachments/assets/f2944d36-1873-4914-a50b-1e3d5114d79a)

**Comment by felixguendling** [comment, marked as answer]
![image](https://github.com/user-attachments/assets/3d63e0e2-9ace-4262-adcd-55bd56462bed)

This is the memory requirement for your setup at the moment. The large 4 GB block comes from having two bits per OSM node. For small datasets, a hashmap would be sufficient and only use a few hundred instead. So there's room to optimize a bit. For now, you probably need a 8 GB RAM server to import and then a 1 GB RAM server to run MOTIS.

For the `resize error` I guess there's either not enough space on disk or in memory. For Windows I would recommend using the Windows binary and run it directly (no VM).

**Comment by NishilJ** [comment]
Thanks I was able to build with 4 GB ram and now using 1 GB to run
